common/readline.c

```diff
@@ -48,6 +49,26 @@ char *read_line(FILE *file) {
 	return string;
 }
 
+char *peek_line(FILE *file, int offset, long *position) {
+	long pos = ftell(file);
+	size_t length = 1;
+	char *line = calloc(1, length);
```

Comment:
No need to `calloc`, you can set `length = 0` and `line = NULL` and `getline` will do that for you.